hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBTable.java

```diff
@@ -57,6 +57,7 @@
   private final ColumnFamilyHandle handle;
   private final WriteOptions writeOptions;
   private final RDBMetrics rdbMetrics;
+  private int prefixLength;
```

Comment:
We don't need to have _prefixLength_ associated with RDBTable. Prefix is something which should be passed as an argument. It is not applicable at table level.

For _getRangeKVs_ calls we can use _MetadataKeyFilters.MetadataKeyFilter_ for passing prefix to be matched.

If we need the Iterator optimized prefix filter to be used, we can add a new _getSequentialRangeKVs_ method which also takes prefix as an argument. (Since we cannot use _MetadataKeyFilters.MetadataKeyFilter_ as prefix filter for Iterator creation)

Reply:
Yes, it is better to prevent a prefixLength set at the table level.

Here I tried to have unified interface for different schemas, so I'll try to use the `MetadataKeyFilters.MetadataKeyFilter` to pass prefix but not introducing new method dedicated for schema v3.

Reply:
OK...I found it not so easy to pass then extract prefix using the `MetadataKeyFilters.MetadataKeyFilter`, because the KeyFilter interface doesn't offer convenient interface for that and hard to refactor. And also we haven't introduced the fixed-length prefix stuff for schemaV3 keys and we can't assume the char encoding in general purpose classes like `MetadataKeyFilters.MetadataKeyFilter` or `RDBTable`.

I'm trying to add a new parameter `prefix` to the `getRangeKVs` & `getSequentialRangeKVs` interfaces. 
I feel that it is painful to do so, but I would like to have abstracted interfaces for all schemas instead of `if-else` checks.